status/statusgo_backend_new/communities.nim

```diff
@@ -9,4 +9,200 @@ proc getJoinedComunities*(): RpcResponse[JsonNode] {.raises: [Exception].} =
+      "position": position
+    }])
+
+proc reorderCommunityChat*(
+    communityId: string,
+    categoryId: string,
+    chatId: string,
+    position: int
+    ): RpcResponse[JsonNode] {.raises: [Exception].} =
+  result = callPrivateRPC("reorderCommunityChat".prefix, %*[
+    {
+      "communityId": communityId,
+      "categoryId": categoryId,
+      "chatId": chatId,
+      "position": position
+    }])
+
+proc deleteCommunityChat*(
+    communityId: string,
+    chatId: string
+    ): RpcResponse[JsonNode] {.raises: [Exception].}  =
+  result = callPrivateRPC("deleteCommunityChat".prefix, %*[communityId, chatId])
+
+proc createCommunityCategory*(
+    communityId: string,
+    name: string,
+    channels: seq[string]
+    ): RpcResponse[JsonNode] {.raises: [Exception].} =
+  result = callPrivateRPC("createCommunityCategory".prefix, %*[
+    {
+      "communityId": communityId,
+      "categoryName": name,
+      "chatIds": channels
+    }])
+
+proc editCommunityCategory*(
+    communityId: string,
+    categoryId: string,
+    name: string,
+    channels: seq[string]
+    ): RpcResponse[JsonNode] {.raises: [Exception].} =
+  result = callPrivateRPC("editCommunityCategory".prefix, %*[
+    {
+      "communityId": communityId,
+      "categoryId": categoryId,
+      "categoryName": name,
+      "chatIds": channels
+    }])
+
+proc deleteCommunityCategory*(
+    communityId: string,
+    categoryId: string
+    ): RpcResponse[JsonNode] {.raises: [Exception].} =
+  result = callPrivateRPC("deleteCommunityCategory".prefix, %*[
+    {
+      "communityId": communityId,
+      "categoryId": categoryId
+    }])
+
+proc requestCommunityInfo*(communityId: string): RpcResponse[JsonNode] {.raises: [Exception].} =
+  result = callPrivateRPC("requestCommunityInfoFromMailserver".prefix, %*[communityId])
+
+proc importCommunity*(communityKey: string): RpcResponse[JsonNode] {.raises: [Exception].} =
+  result = callPrivateRPC("importCommunity".prefix, %*[communityKey])
+
+proc exportCommunity*(communityId: string): RpcResponse[JsonNode] {.raises: [Exception].}  =
+  result = callPrivateRPC("exportCommunity".prefix, %*[communityId])
+
+proc removeUserFromCommunity*(communityId: string, pubKey: string) RpcResponse[JsonNode] {.raises: [Exception].}  =
+  result = callPrivateRPC("removeUserFromCommunity".prefix, %*[communityId, pubKey])
+
+proc requestToJoinCommunity*(communityId: string, ensName: string): RpcResponse[JsonNode] {.raises: [Exception].}  =
+  result = callPrivateRPC("requestToJoinCommunity".prefix, %*[{
+    "communityId": communityId,
+    "ensName": ensName
+  }])
+
+  var communityRequests: seq[CommunityMembershipRequest] = @[]
+  if rpcResult{"result"}{"requestsToJoinCommunity"} != nil and rpcResult{"result"}{"requestsToJoinCommunity"}.kind != JNull:
+    for jsonCommunityReqest in rpcResult["result"]["requestsToJoinCommunity"]:
+      communityRequests.add(jsonCommunityReqest.toCommunityMembershipRequest())
+
+  return communityRequests
```

Comment:
@borismelnik as I see here you're returning an array of `CommunityMembershipRequest `, but the returned type for this method is `RpcResponse[JsonNode]`, anyway this logic shouldn't be here.